enhancements/network/sdn-live-migration.md

```diff
@@ -0,0 +1,404 @@
+  then just routes it normally.
+- Ingress NetworkPolicy processing will happen when the packet reaches the
+  destination pod's switch port, just like normal.
+  - Our NetworkPolicy rules are all based on IP addresses, not "logical input
+    port", etc, so it doesn't matter that the packets came from outside OVN and
+    have no useful OVN metadata.
+
+#### Packets going from OVN Kubernetes to OpenShift SDN
+
+On the ovn side:
+- The packet just follow the same path of hybrid overlay. The packet just has to
+  get routed out the VXLAN tunnel with VNID 0.
+
+On the sdn side:
+- We have to change ingress NetworkPolicy processing to be based entirely on pod
+  IPs rather than using namespace VNIDs, since packets from ovn nodes won"t have
+  the VNID set. There is already code to generate the rules that way though,
+  because egress NetworkPolicy already works that way.
+
+### Workflow Description
+
+#### Migration Setup
+
+1. The admin kicks off the migration process by updating the custom resource
+   `network.operator`:
+   
+   ```bash
+   $ oc patch Network.operator.openshift.io cluster --type='merge' --patch '{"spec":{"migration":{"networkType":"OVNKubernetes","isLive":true}}}"
+   ```
+   
+   CNO will check if there is any unsupported feature (refer the
+   [Limitations](#limitations) section) enabled in the cluster. If yes, it will
+   set an warning message under the `status.migration` field of the custom
+   resource `network.operator`.
+
+2. CNO will redeploy the openshift-sdn in the __migration mode__. It will add a
+   condition check to the wrapper script of the sdn DaemonSet, to see if the
+   bridge `br-ex` exists on the node . If `br-ex` exists, it means the node has
+   already been updated by MCO, therefore it's ready for running OVN Kubernetes,
+   the sdn pod would just do "sleep infinity" rather than actually launching the
+   `openshift-sdn-node` process.
+
+3. CNO will also deploy OVN Kubernetes to the cluster with hybrid overlay
+   enabled. The ovnkube-node DaemonSet wrapper script also has the logic to
+   check if the bridge br-ex exists on the node. If `br-ex` doesn't exist, it
+   means the node has yet been updated by MCO, so instead of starting the
+   `ovnkube-node` process, it would run `hybrid-overlay-node` process that
+   discover the necessary hybrid overlay node information and annotate node
+   accordingly e.g.:
+
+   ```yaml
+   k8s.ovn.org/hybrid-overlay-distributed-router-gateway-mac: 00-c2-f5-92-28-ad
+   k8s.ovn.org/hybrid-overlay-node-subnet: 192.168.111.20/24
+   ```
+
+   otherwise it would launch the `ovnkube-node` process.
+
+#### Migration
+
+We can kick off the migration from CNO, by:
+
+```bash
+$ oc patch Network.config.openshift.io cluster --type='merge' --patch '{"spec":{"networkType":"OVNKubernetes"}'
+```
+
+CNO will update the `status.networkType` field of the `network.config` CR. It
+will trigger MCO to apply new MachineConfig to each node:
```

Comment:
what new MachineConfig?

Reply:
The new MachineConfig is rendered for running OVN-K.

Reply:
ok but you haven't mentioned any MachineConfig existing or being created yet, or what it contains, etc

Reply:
I added more details.